assets/blocks/quiz/question-block/question-block-attributes.js

```diff
@@ -34,29 +48,149 @@ export function createQuestionBlockAttributes( question ) {
 
 /**
  * Type-specific block attributes.
+ *
+ * @param {Object} apiAttributes The REST API response for a question.
  */
-const getTypeAttributes = {
-	'gap-fill': ( { before, after, gap } ) => ( {
-		answer: {
-			textBefore: before,
-			textAfter: after,
-			rightAnswers: gap,
-		},
-	} ),
-	'multiple-choice': ( { options, random_order: randomOrder } ) => ( {
-		answer: {
-			answers: options.map( ( { label: title, correct: isRight } ) => ( {
-				title,
-				isRight,
-			} ) ),
-		},
-		options: {
-			randomOrder,
-		},
-	} ),
-	boolean: ( { answer: rightAnswer } ) => ( {
-		answer: {
-			rightAnswer,
-		},
-	} ),
+const getTypeAttributes = ( apiAttributes ) => {
+	let blockAttributes;
+
+	switch ( apiAttributes.type ) {
+		case 'gap-fill':
+			blockAttributes = {
+				answer: {
+					textBefore: apiAttributes.before,
+					textAfter: apiAttributes.after,
+					rightAnswers: apiAttributes.gap,
+				},
+			};
+			break;
+		case 'multiple-choice':
+			blockAttributes = {
+				answer: {
+					answers: apiAttributes.options.map(
+						( { label: title, correct: isRight } ) => ( {
+							title,
+							isRight,
+						} )
+					),
+				},
+				options: {
+					randomOrder: apiAttributes.random_order,
+				},
+			};
+			break;
+		case 'boolean':
+			blockAttributes = {
+				answer: {
+					rightAnswer: apiAttributes.answer,
+				},
+			};
+			break;
+		case 'single-line':
+		case 'multi-line':
+		case 'file-upload':
+			blockAttributes = {};
+			break;
+		default:
+			blockAttributes = {
+				answer: normalizeAttributes( apiAttributes, camelCase ),
+			};
+			break;
+	}
+
+	/**
+	 * Filters the attributes that will be supplied to the question block after they are received by REST API.
+	 *
+	 * @param {string} type            The question type.
+	 * @param {Object} blockAttributes The block attributes.
```

Comment:
Think these might be flipped

Reply:
Good catch! Fixed in 13bcec6f37303556763cb53961050a384b65666c